src/main/java/org/apache/sysml/api/python/SystemML.py

```diff
@@ -247,3 +257,163 @@ def getStringRDD(self, varName, format):
         #    return rdd
         #except Py4JJavaError:
         #    traceback.print_exc()
+
+def getNumCols(numPyArr):
+	if len(numPyArr.shape) == 1:
+		return 1
+	else:
+		return numPyArr.shape[1]
+       
+def convertToJavaMatrix(sc, src):
+	if isinstance(src, np.ndarray):
+		from array import array
+		numCols = getNumCols(src)
+		numRows = src.shape[0]
+		if src.dtype.type is np.float64:
+			arr = src.reshape(-1)
+		else:
+			arr = array('d', src.reshape(-1))
+		buf = bytearray(arr.tostring())
+		return sc._jvm.org.apache.sysml.runtime.instructions.spark.utils.RDDConverterUtilsExt.convertPy4JArrayToMB(buf, numRows, numCols)
+	else:
+		raise Exception('Type is not supported')
+
+def convertToNumpyArr(sc, mb):
+	numRows = mb.getNumRows()
+	numCols = mb.getNumColumns()
+	buf = sc._jvm.org.apache.sysml.runtime.instructions.spark.utils.RDDConverterUtilsExt.convertMBtoPy4JDenseArr(mb)
+	return np.frombuffer(buf, count=numRows*numCols, dtype=np.float64)
+
+class mllearn:
+    # Or we can create new Python project with package structure
+    class LogisticRegression(Estimator):
+
+        def __init__(self, sqlCtx, penalty='l2', fit_intercept=True, max_iter=100, max_inner_iter=0, tol=0.000001, C=1.0, solver='newton-cg', transferUsingDF=False):
+            self.sqlCtx = sqlCtx
+            self.sc = sqlCtx._sc
+            self.log = self.sc._jvm.org.apache.sysml.api.ml.LogisticRegression("lr", self.sc._jsc.sc())
+            self.transferUsingDF = transferUsingDF
+            if penalty != 'l2':
+                raise Exception('Only l2 penalty is supported')
+            if fit_intercept:
+                self.icpt = 1
+            else:
+                self.icpt = 0
+            self.max_iter = max_iter
+            self.max_inner_iter = max_inner_iter
+            self.tol = tol
+            if C == 0:
+                raise Exception('C cannot be 0')
+            reg = 1/C
+            self.reg = reg
+            self.updateLog()
+            if solver != 'newton-cg':
+                raise Exception('Only newton-cg solver supported')
+             
+        def updateLog(self):
+            self.log.setMaxOuterIter(self.max_iter)
+            self.log.setMaxInnerIter(self.max_inner_iter) 
+            self.log.setRegParam(self.reg)
+            self.log.setTol(self.tol)
+            self.log.setIcpt(self.icpt)
+            
+        def convertToPDF(self, X):
```

Comment:
A slightly more explicit name might be better. It kind of reminds me of [this](https://www.freepdfconvert.com/) ;)